src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -4035,10 +4037,18 @@ private void ParseParameterModifiers(SyntaxListBuilder modifiers)
                 {
                     case SyntaxKind.ThisKeyword:
                         modifier = CheckFeatureAvailability(modifier, MessageID.IDS_FeatureExtensionMethod);
+                        if (seenRefModifier)
+                        {
+                            modifier = CheckFeatureAvailability(modifier, MessageID.IDS_FeatureRefExtensionMethods);
```

Comment:
cc @VSadov do we have plans on grouping them? right now, this change can only be affecting another feature (readonly references), which isn't a 100% overlap since you can have ref extension methods.